From d5fc0ea7b8edde511a8de40b7e0113017f51e482 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Fri, 4 Jan 2013 17:47:17 +0800
Subject: [PATCH 5/7] ENGR00238882: mtd-utils: make pass build and add libubi
 static lib.

JB4.2 use google toolchain, which leak of many tool and lib, use
server 's toolchain to pass build.

Also add a static lib for libubi which can be used for recovery.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 mkfs.ubifs/Android.mk |  5 ++++-
 new-utils/Android.mk  | 21 +++++++++++++++++++++
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/mkfs.ubifs/Android.mk b/mkfs.ubifs/Android.mk
index 512b562..8fc1536 100644
--- a/mkfs.ubifs/Android.mk
+++ b/mkfs.ubifs/Android.mk
@@ -14,16 +14,19 @@ LOCAL_SRC_FILES := \
 	hashtable/hashtable.c \
 	hashtable/hashtable_itr.c
 
+LOCAL_CC := /usr/bin/gcc
+LOCAL_CXX := /usr/bin/g++
 LOCAL_CFLAGS = -O2 -Wall
 LOCAL_CFLAGS+= -Wpointer-arith -Wwrite-strings -Wstrict-prototypes
 LOCAL_CFLAGS+= -Wmissing-prototypes -Winline
+LOCAL_LDFLAGS += -L/usr/lib/x86_64-linux-gnu
 LOCAL_LDLIBS:= -lz -llzo2 -lm -luuid
 
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/
+LOCAL_C_INCLUDES += /usr/include/
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/hashtable
 
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := mkfs.ubifs
 
 include $(BUILD_HOST_EXECUTABLE)
-
diff --git a/new-utils/Android.mk b/new-utils/Android.mk
index 9a677b2..59b8286 100644
--- a/new-utils/Android.mk
+++ b/new-utils/Android.mk
@@ -46,6 +46,27 @@ LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_SHARED_LIBRARY)
 
+# libubi static lib.
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := \
+	src/libubi.c \
+	src/libmtd.c \
+	src/libscan.c \
+	src/libubigen.c \
+	src/crc32.c \
+	src/common.c
+
+LOCAL_CFLAGS = -O2 -Wall
+LOCAL_CFLAGS += -Wall -Wextra -Wwrite-strings -Wno-sign-compare -D_FILE_OFFSET_BITS=64
+
+LOCAL_C_INCLUDES += $(LOCAL_PATH)/include
+LOCAL_C_INCLUDES += $(LOCAL_PATH)/src
+
+LOCAL_MODULE := libubi
+LOCAL_MODULE_TAGS := optional
+include $(BUILD_STATIC_LIBRARY)
+
 # ubinfo util
 #
 include $(CLEAR_VARS)
-- 
1.8.0

